src/libstd/sys/windows/fs.rs

```diff
@@ -377,8 +377,13 @@ impl fmt::Debug for File {
     }
 }
 
-pub fn to_utf16(s: &Path) -> Vec<u16> {
-    s.as_os_str().encode_wide().chain(Some(0)).collect()
+pub fn to_u16s(s: &Path) -> io::Result<Vec<u16>> {
```

Comment:
Ah actually I was digging around the other day and ran across [another `to_utf16` function](https://github.com/rust-lang/rust/blob/3e48b0e380319dc586a329baac640b9457feb87a/src/libstd/sys/windows/mod.rs#L75), could you merge this with that and update the signature there as well?